src/partition/repart.c

```diff
@@ -1699,6 +1702,59 @@ static int context_dump_partitions(Context *context, const char *node) {
         return 0;
 }
 
+static int generate_report(Context *context, const char *node) {
```

Comment:
hmm, so it was my hope that we could just reuse the table that context_dump_partitions() outputs, and maybe add a few more columns that we incude in the JSON table but maybe turn off in the text version (i.e. using text_set_display())

Reply:
There are two reasons why I did not do that:
* The current table has graphical elements that do provide value in the output and that I would want to keep. So I would need to create a different table for JSON anyway (e.g. with separate columns for old_value and new_value instead of merging both values into one string with an arrow between.
* The existing table summarizes the changes systemd-repart *wants* to do, while I want to have a report of changes that systemd-repart actually *did*.

Reply:
The table is way more flexible then I first understood and the return value is a good indicator of whether or not systemd-repart succeeded, so my arguments are void.

I changed this now.